### PR TITLE
OG-799 log fix OG-803 eip1559 detection fix

### PR DESCRIPTION
**Pull request by shahafn**
Also minor defaultPriorityFee fix

### CODE REVIEW

File: packages/common/src/ContractInteractor.ts [hunk 1]
```diff
@@ -216,9 +216,17 @@ export class ContractInteractor {
...
       throw new Error('init was already called')
     }
-    const block = await this.web3.eth.getBlock('latest').catch((e: Error) => { throw new Error(`getBlock('latest') failed: ${e.message}\nCheck your internet/ethereum node connection`) })
-    if (block.baseFeePerGas != null) {
+    await this.web3.eth.getBlock('latest').catch((e: Error) => { throw new Error(`getBlock('latest') failed: ${e.message}\nCheck your internet/ethereum node connection`) })
+    try {
+      await this.getFeeHistory('0x1', 'latest', [0.5])
       this.transactionType = TransactionType.TYPE_TWO
+    } catch (e: any) {
+      if ((e as Error).message.includes('the method eth_feeHistory does not exist/is not available')) {
```
Comment (drortirosh) [inline packages/common/src/ContractInteractor.ts:224]: https://api.avax.network/ext/bc/C/rpc	{"code":-32601,"message":"the method eth_methodName does not exist"}
https://bsc-dataseed1.ninicoin.io	{"code":-32601,"message":"the method eth_methodName does not exist/is not available"}
https://cloudflare-eth.com	{"code":-32601,"message":"Method not found"}
https://polygon-rpc.com	{"code":-32601,"message":"the method eth_methodName does not exist/is not available"}
https://arb1.arbitrum.io/rpc	{"code":-32601,"message":"the method eth_methodName does not exist/is not available"}
https://mainnet.optimism.io	{"code":-32001,"message":"rpc method is not whitelisted"}
https://dai.poa.network	{"code":-32601,"message":"Method eth_methodName is not supported"}

Comment (shahafn) [reply packages/common/src/ContractInteractor.ts:224]: Ok, screening by code then.

Comment (drortirosh) [reply packages/common/src/ContractInteractor.ts:224]: https://mainnet.optimism.io	{"code":-32001,"message":"rpc method is not whitelisted"}

Comment (drortirosh) [reply packages/common/src/ContractInteractor.ts:224]: the list is more problematic than we thought. Here I test for actual method, instead for dummy "eth_methodName"):

1. arbitrum, optimism don't support it yet, which is strange for me. need to ask (they should use geth - don't they?)
2. see that infura and cloudflare - both eth mainnet - have different results..

https://mainnet.infura.io/v3/c9ea2fff9f9342d0b8bea654f890b56e:	{"jsonrpc":"2.0","id":1234,"error":{"code":-32602,"message":"missing value for required argument 0"}}
https://cloudflare-eth.com:	{"jsonrpc":"2.0","error":{"code":-32601,"message":"Method not found"},"id":1234}
https://bsc-dataseed2.ninicoin.io:	{"jsonrpc":"2.0","id":1234,"error":{"code":-32601,"message":"the method eth_feeHistory does not exist/is not available"}}
https://rpc.ankr.com/bsc:	{"jsonrpc":"2.0","id":1234,"error":{"code":-32601,"message":"the method eth_feeHistory does not exist/is not available"}}
https://api.avax.network/ext/bc/C/rpc:	{"jsonrpc":"2.0","id":1234,"error":{"code":-32602,"message":"missing value for required argument 0"}}
https://rpc.ankr.com/avalanche:	{"jsonrpc":"2.0","id":1234,"error":{"code":-32602,"message":"missing value for required argument 0"}}
https://polygon-rpc.com:	{"jsonrpc":"2.0","id":1234,"error":{"code":-32602,"message":"missing value for required argument 0"}}
https://rpc-mainnet.matic.network:	{"jsonrpc":"2.0","id":1234,"error":{"code":-32602,"message":"missing value for required argument 0"}}
https://evm-cronos.crypto.org:	{"jsonrpc":"2.0","id":1234,"error":{"code":-32602,"message":"missing value for required argument 0"}}
https://evm.cronos.org:	{"jsonrpc":"2.0","id":1234,"error":{"code":-32602,"message":"missing value for required argument 0"}}
https://rpc2.fantom.network:	{"jsonrpc":"2.0","id":1234,"error":{"code":-32602,"message":"missing value for required argument 0"}}
https://rpc.ftm.tools:	{"jsonrpc":"2.0","id":1234,"error":{"code":-32601,"message":"the method eth_feeHistory does not exist/is not available"}}
https://arb1.arbitrum.io/rpc:	{"jsonrpc":"2.0","id":1234,"error":{"code":-32601,"message":"the method eth_feeHistory does not exist/is not available"}}
https://rpc.ankr.com/arbitrum:	{"jsonrpc":"2.0","id":1234,"error":{"code":-32601,"message":"the method eth_feeHistory does not exist/is not available"}}
https://mainnet.optimism.io:	{"jsonrpc":"2.0","error":{"code":-32001,"message":"rpc method is not whitelisted"},"id":1234}
https://optimism.gnosischain.com:	{"jsonrpc":"2.0","id":1234,"error":{"code":-32601,"message":"the method eth_feeHistory does not exist/is not available"}}
https://dai.poa.network:	{"jsonrpc":"2.0","error":{"code":-32602,"message":"Invalid params","data":"Incorrect parameters count, expected: 3, actual: 0"},"id":1234}
https://mycrypto.rsk.co:	{"jsonrpc":"2.0","id":1234,"error":{"code":-32601,"message":"method not found"}}
https://public-node.rsk.co:	{"jsonrpc":"2.0","id":1234,"error":{"code":-32601,"message":"method not found"}}

Comment (shahafn) [reply packages/common/src/ContractInteractor.ts:224]: 1. Please don't publish infura ids on github
2. mainnet and cloudflare differ becasue these are different errors. Just send a valid request with the correct parameters.

**Review by forshtat — APPROVED** [review]

**Pull request merged**